rejistry++/src/RegistryHiveBuffer.cpp

```diff
@@ -47,7 +49,10 @@ namespace Rejistry {
     }
 
     RegistryKey * RegistryHiveBuffer::getRoot() const {
-        return new RegistryKey(getHeader()->getRootNKRecord());
+        REGFHeader *header = getHeader();
+        Rejistry::NKRecord *nkRecord = header->getRootNKRecord();
+        delete header;
```

Comment:
I don't believe getHeader() made a copy of the header, so it probably shouldn't be freed here

Reply:
getHeader() return new REGFHeader, line 59
    
    REGFHeader * RegistryHiveBuffer::getHeader() const {
        return new REGFHeader(*_buffer, 0x0);
    }

Reply:
Yes I know there's a new there. But trace that down:
REGFHeader::REGFHeader(RegistryByteBuffer& buf, const uint32_t offset) : BinaryBlock(buf, offset) 


        BinaryBlock(RegistryByteBuffer& buf, uint32_t offset) {

            _buf = &buf;
            _offset = offset;
        }
I don't see a new copy of that buffer getting made, just a new pointer getting assigned to it.

Reply:
Though I guess delete maybe doesn't delete that buffer? This code seems kind of terrible.

Reply:
Terrible is so correct. Initially I had a hard time understanding the VLD stacktrace.